service/parserprovider/setflag.go

```diff
@@ -47,16 +45,17 @@ func (sfl *setFlagProvider) Get() (*configparser.Parser, error) {
 		return sfl.base.Get()
 	}
 
-	b := &bytes.Buffer{}
+	parsed := map[string]interface{}{}
 	for _, property := range flagProperties {
-		property = strings.TrimSpace(property)
-		if _, err := fmt.Fprintf(b, "%s\n", property); err != nil {
-			return nil, err
+		split := strings.Split(strings.TrimSpace(property), "=")
+		if len(split) != 2 {
+			return nil, fmt.Errorf("failed to parse set flag %q", property)
 		}
+		parsed[split[0]] = split[1]
 	}
-	viperFlags := viper.NewWithOptions(viper.KeyDelimiter(configparser.KeyDelimiter))
-	viperFlags.SetConfigType(setFlagFileType)
-	if err := viperFlags.ReadConfig(b); err != nil {
+
+	propertyKoanf := koanf.New(".")
+	if err := propertyKoanf.Load(confmap.Provider(parsed, "."), nil); err != nil {
 		return nil, fmt.Errorf("failed to read set flag config: %v", err)
 	}
```

Comment:
Not sure this is as simple as a parsing with "." (it may be, but I want you to confirm/check that).
Viper is using 	"github.com/magiconair/properties" not just a "string split", so why would they use that library instead of simply splitting?

Reply:
Still looking into this.

Reply:
Not sure if gh lost my comment or I never submitted:

I missed that magiconair/properties properties.Get(key) will use env var expansion, so am recreating underlying logic while using that lib directly.